File: src/main/java/ch/qligier/jetbrains/plugin/fhir/fsh/psi/DebugPsiNode.java
// Copyright 2024 Quentin Ligier. Use of this source code is governed by the MIT license.

package ch.qligier.jetbrains.plugin.fhir.fsh.psi;

import com.intellij.lang.ASTNode;
import org.antlr.intellij.adaptor.psi.ANTLRPsiNode;
import org.jetbrains.annotations.NotNull;

/**
 * jetbrains-plugin-fhir
 *
 * @author Quentin Ligier
 **/
public class DebugPsiNode extends ANTLRPsiNode {

    private final String text;
    private final ASTNode node;

    public DebugPsiNode(@NotNull final ASTNode node,
                        final String text) {
        super(node);
        this.text = text;
        this.node = node;
    }

    @Override
    public String toString() {
        return getClass().getSimpleName() + "(" + node.getElementType() + ", " + text + ")";
    }
}

File: src/main/kotlin/ch/qligier/jetbrains/plugin/fss/fsh/psi/DebugPsiNode.kt
// Copyright 2025 Quentin Ligier. Use of this source code is governed by the MIT license.
package ch.qligier.jetbrains.plugin.fss.fsh.psi

import com.intellij.lang.ASTNode
import org.antlr.intellij.adaptor.psi.ANTLRPsiNode

class DebugPsiNode(
    private val node: ASTNode,
    private val text: String?
) : ANTLRPsiNode(node) {
    override fun toString(): String {
        return javaClass.getSimpleName() + "(" + node.elementType + ", " + text + ")"
    }
}
